src/argilla/client/feedback/schemas.py

```diff
@@ -93,51 +93,165 @@ def user_id_must_have_value(cls, v):
+        ...            question_name="question-1",
+        ...            type="model",
+        ...            score=0.9,
+        ...            value="This is the first suggestion",
+        ...            agent="agent-1",
+        ...         ),
         ...     external_id="entry-1",
         ... )
 
     """
 
+    id: Optional[UUID] = None
     fields: Dict[str, str]
-    metadata: Optional[Dict[str, Any]] = None
-    responses: Optional[Union[ResponseSchema, List[ResponseSchema]]] = None
+    metadata: Optional[Dict[str, Any]] = Field(default_factory=dict)
+    responses: Optional[List[ResponseSchema]] = Field(default_factory=list)
+    suggestions: Optional[List[SuggestionSchema]] = Field(default_factory=list, allow_mutation=True)
     external_id: Optional[str] = None
-    _unified_responses: Optional[Dict[str, List["UnifiedValueSchema"]]] = PrivateAttr(default={})
 
-    @validator("responses", always=True)
-    def responses_must_be_a_list(cls, v: Optional[Union[ResponseSchema, List[ResponseSchema]]]) -> List[ResponseSchema]:
-        if not v:
-            return []
-        if isinstance(v, ResponseSchema):
-            return [v]
-        return v
+    _unified_responses: Optional[Dict[str, List["UnifiedValueSchema"]]] = PrivateAttr(default_factory=dict)
+    _updated: bool = PrivateAttr(default=False)
+
+    def __init__(self, **data: Dict[str, Any]) -> None:
+        super().__init__(**data)
+
+    def add_suggestions(self, suggestions: Union[SuggestionSchema, List[SuggestionSchema]]) -> None:
+        if isinstance(suggestions, (dict, SuggestionSchema)):
+            suggestions = [suggestions]
+        parsed_suggestions = []
+        for suggestion in suggestions:
+            if not isinstance(suggestion, SuggestionSchema):
+                suggestion = SuggestionSchema(**suggestion)
+            parsed_suggestions.append(suggestion)
+        if not self.id:
+            warnings.warn(
+                "Ignore the following if you are creating a new `FeedbackDataset` with"
+                " `FeedbackRecord`s, or if you are just working with a `FeedbackRecord`."
+                " Otherwise, if the `FeedbackRecord` is already pushed"
+                " to Argilla, note that `suggestions` have been provided, but the `id`"
+                " is not set, which means that the `FeedbackRecord` has been pushed to"
+                " Argilla, but hasn't been fetched, so the `id` is missing. To solve that,"
+                " you can simply call `FeedbackDataset.fetch_records()` to fetch them and"
+                " automatically set the `id`, to call `add_suggestions` on top of that."
+            )
+        for suggestion in parsed_suggestions:
+            existing_suggestion = next(
+                filter(
+                    lambda s: (s.question_id == suggestion.question_id)
+                    or (s.question_name == suggestion.question_name),
+                    self.suggestions,
+                ),
```

Comment:
I would use a list comprehension as I think it's more readable

Reply:
just as an idea. could we just define record suggestions as a dictionary where the key is the question name and the value is the suggestion itself?

Reply:
I also feel a bit hard to read this block of code

Reply:
Note that for the suggestions we cannot do that as we may also want to add `type`, `score` and `agent`, so I included the `question_name` and the mapping to `question_id` before pushing to Argilla to make it easier for the user

Reply:
That block of code is an iterator that checks for each entry that condition until is matched, otherwise returns None, but if you both feel like a for-loop is better I'll revert